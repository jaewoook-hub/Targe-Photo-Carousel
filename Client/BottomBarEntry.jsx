import React from 'react'
import styles from './BottomBar.module.css';

class BottomBarEntry extends React.Component {
  constructor(props) {
    super(props)
  }

  handleClick() {
    var imageIndex = this.props.imageList.indexOf(this.props.image);
    this.props.changeMainImage(this.props.image._id, imageIndex)
  }

  render () {

    return (
      <div >
        <img
          className={styles.BottomBarEntry}
          src={this.props.image.url}
          alt={this.props.image.alt}
          onClick={this.handleClick.bind(this)}
          >
          </img>
      </div>
    )
  }
}
export default BottomBarEntry;
