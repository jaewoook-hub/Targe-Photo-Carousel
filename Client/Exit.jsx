import React from 'react';
import styles from './Button.module.css';

class Exit extends React.Component {
  constructor(props) {
    super(props)
  }

  handleClick() {
    this.props.changeViewBack()
  }

  render() {
    return (
      <div>
        <button className={styles.exitBTN} onClick={this.handleClick.bind(this)}>X</button>
      </div>
    )
  }
}

export default Exit;
