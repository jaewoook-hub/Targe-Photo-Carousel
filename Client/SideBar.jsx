import React from 'react';
import SideBarEntry from './sideBarEntry.jsx';
import GreyedOutImage from './GreyedOutImage.jsx'
import styles from './SideBar.module.css'

const SideBar = (props) => (
<div className={styles.sideBarContainer}>
   {props.imageList.map((image) =>
      <SideBarEntry
        image={image}
        changeMainImage={props.changeMainImage}
        key={image._id}
        imageList={props.imageList}
      />
    )}
    <GreyedOutImage
      extendImage={props.extendImage}
      clickMainImageForPopOut={props.clickMainImageForPopOut}
      mainImageId={props.mainImageId}
      imageList={props.imageList}
      imageListLength={props.imageListLength}/>
</div>
)
export default SideBar;
