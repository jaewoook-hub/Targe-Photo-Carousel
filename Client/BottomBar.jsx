import React from 'react';
import BottomBarEntry from './BottomBarEntry.jsx'
import styles from './BottomBar.module.css';

const BottomBar = (props) => (
      <div className={styles.bottomBar}>
        {props.imageList.map((image) => <BottomBarEntry
          image={image}
          changeMainImage={props.changeMainImage}
          imageList={props.imageList}
          key={image._id}/>
      )}</div>
)

export default BottomBar;
