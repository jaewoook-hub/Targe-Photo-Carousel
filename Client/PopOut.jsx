import React from 'react';
import BigImage from './PopOutBigImage.jsx';
import Exit from './Exit.jsx'
import BottomBar from './BottomBar.jsx'
import NextPicture from './NextPictureButton.jsx'
import LastPicture from './LastPictureButton.jsx'
import styles from './PopOut.module.css';


const PopOut = (props) => (
          <div className="popOut">
            <div className={styles.bigCarousel}>
              <LastPicture lastImage={props.lastImage}/>
              <BigImage mainImage={props.mainImage}/>
              <div>
              <Exit changeViewBack={props.changeViewBack}/>
              <NextPicture nextImage={props.nextImage}/>

              </div>
            </div>
            <div>
              <BottomBar imageList={props.imageList}
                         changeMainImage={props.changeMainImage}/>
            </div>
          </div>
)
  export default PopOut;
