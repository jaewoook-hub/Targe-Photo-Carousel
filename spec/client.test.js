import 'jsdom-global/register';
import React from 'react';
import { shallow } from 'enzyme';
import ImageCarousel from '../Client/App.jsx';
import SideBar from '../Client/SideBar.jsx';
import SideBarEntry from '../Client/sideBarEntry.jsx';
import { mount } from 'enzyme';
import MainImage from '../Client/mainImage.jsx';
const faker = require('faker');
import renderer from 'react-test-renderer';
import PopOut from '../Client/PopOut.jsx';
import BigImage from '../Client/PopOutBigImage.jsx';
import BottomBar from '../Client/BottomBar.jsx';
import BottomBarEntry from '../Client/BottomBarEntry.jsx';
import Exit from '../Client/Exit.jsx';

test('truth serum', () => {
  expect('true').toBe('true');
});

var mockImageDataCreator = function(numberOfEnries) {

    var dataEntries = [];
    for (var i = 0; i < numberOfEnries; i++) {
      dataEntries.push({
         product: faker.commerce.productName().split(' ').join('-'),
         imageName: faker.commerce.productDescription(),
         color: faker.commerce.color().split(' ').join('-'),
         url: 'https://target-image-carousel.s3-us-west-1.amazonaws.com/Screen+Shot+2020-09-24+at+5.21.25+PM.png',
         alt: faker.image.abstract()
      })
    }
    return dataEntries;
  }

  var mockImages = mockImageDataCreator(7);
  var mockSideBarImages = mockImages.slice(0, 4)
  var mockOneImage = mockImages[0];
  var mockExtendImage = mockImages.slice(4, 5)[0]

describe('ImageCarousel', () => {

  test('should render the ImageCarousel component correctly', () => {
    const wrapper = shallow(<ImageCarousel />);

    expect(wrapper).toMatchSnapshot();
  })
  test('should render 1 SideBar', () => {
    const wrapper = shallow(<SideBar imageList={mockSideBarImages}/>);
    expect(wrapper.exists()).toBe(true);
  })

  test('should render one main image', () => {
    const wrapper = mount(<MainImage image={mockOneImage}/>);
    expect(wrapper.find('MainImage')).toHaveLength(1);
  })
})


describe('SideBar', () => {
  test('should render 4 sidebar entry image components', () => {
    const wrapper = mount(<SideBar imageList={mockSideBarImages} extendImage={mockExtendImage}/>);
    expect(wrapper.find(SideBarEntry)).toHaveLength(4);
  })
  test('each sidebar entry should have an img tag', () => {
    const wrapper = mount (<SideBarEntry image={mockOneImage}/>);
   expect(wrapper.find('img')).toHaveLength(1)
  })

})


describe('PopOut', () => {
  test('should render the PopOut component correctly', () => {
    const wrapper = shallow(<PopOut />);

    expect(wrapper).toMatchSnapshot();
  })
  test('should render 1 big image', () => {
    const wrapper = shallow(<PopOut />);
    expect(wrapper.find(BigImage)).toHaveLength(1)
  })
  test('should render 1 bottom bar', () => {
    const wrapper = shallow(<PopOut />);
    expect(wrapper.find(BottomBar)).toHaveLength(1)
  })
  test('should render bottom bar entry for each in image list', () => {
    const wrapper = shallow(<BottomBar imageList={mockImages}/>);
    expect(wrapper.find(BottomBarEntry)).toHaveLength(mockImages.length)
  })
  test('each sidebar entry should have an img tag', () => {
    const wrapper = mount (<BottomBarEntry image={mockOneImage}/>);
   expect(wrapper.find('img')).toHaveLength(1)
  })
  test('should conatin one exit component', () => {
    const wrapper = shallow(<PopOut />);
    expect(wrapper.find(Exit)).toHaveLength(1);
  })
})
