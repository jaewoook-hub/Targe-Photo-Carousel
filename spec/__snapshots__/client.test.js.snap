// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel should render the ImageCarousel component correctly 1`] = `<div />`;

exports[`PopOut should render the PopOut component correctly 1`] = `
<div
  className="popOut"
>
  <div>
    <LastPicture />
    <BigImage />
    <div>
      <Exit />
      <NextPicture />
    </div>
  </div>
  <div>
    <BottomBar />
  </div>
</div>
`;
